### PR TITLE
Move process list to Rust Host and delete HostCInternal

### PR DESCRIPTION
**Pull request by sporksmith**

### CODE REVIEW

File: src/main/host/host.rs [hunk 1]
```diff
@@ -188,10 +188,7 @@ impl crate::utility::IsSend for Host {}
 // TODO: use derive(Debug) if/when all fields implement Debug.
 impl std::fmt::Debug for Host {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.debug_struct("Host")
-            .field("chost", &self.chost)
-            .field("info", &self.info)
-            .finish()
+        f.debug_struct("Host").field("info", &self.info).finish()
```
Comment (stevenengler) [inline src/main/host/host.rs:191]: nit: maybe `finish_non_exhaustive` instead since there are fields we aren't printing?

**Review by stevenengler — APPROVED** [review]
🎉

File: src/main/host/process.rs [hunk 2]
```diff
@@ -11,7 +11,7 @@ use super::timer::Timer;
 
 use shadow_shim_helper_rs::HostId;
 
-#[derive(Debug, PartialEq, Eq, Hash, Copy, Clone)]
+#[derive(Debug, PartialEq, Eq, Hash, Copy, Clone, Ord, PartialOrd)]
 pub struct ProcessId(u32);
```
[the remark this answers is not in the document]
Comment (sporksmith) [reply src/main/host/process.rs]: done

File: src/main/host/host.rs (hunk 1 above)
Comment (sporksmith) [reply src/main/host/host.rs]: done

**Pull request merged**